File: php/features.php
<?php
declare(strict_types=1);

// UnsolicitedAdvice SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class UnsolicitedAdviceFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new UnsolicitedAdviceBaseFeature();
            case "test":
                return new UnsolicitedAdviceTestFeature();
            default:
                return new UnsolicitedAdviceBaseFeature();
        }
    }
}

File: py/features.py
# UnsolicitedAdvice SDK feature factory

from feature.base_feature import UnsolicitedAdviceBaseFeature
from feature.test_feature import UnsolicitedAdviceTestFeature


def _make_feature(name):
    features = {
        "base": lambda: UnsolicitedAdviceBaseFeature(),
        "test": lambda: UnsolicitedAdviceTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
